### PR TITLE
[8/?] - lncli+lnrpc: add support for creating unadvertised taproot channels 

### PR DESCRIPTION
**Pull request by Roasbeef**
## Change Description

Depends on https://github.com/lightningnetwork/lnd/pull/7346. 

Only the **last 3 commits** are new. 

In this PR, we update the CLI+RPC interface to allow new taproot channels to be created. 

The next PR in this series will tackle the contract court and the sweeper, to enable full on chain interaction and HTLC resolution for the new channels.

## Steps to Test
Steps for reviewers to follow to test the change.

### Testing
- [ ] Your PR passes all CI checks.
- [ ] Tests covering the positive and negative (error paths) are included.
- [ ] Bug fixes contain tests triggering the bug to prevent regressions.

### Code Style and Documentation
- [ ] The change obeys the [Code Documentation and Commenting](https://github.com/lightningnetwork/lnd/blob/master/docs/code_contribution_guidelines.md#CodeDocumentation) guidelines, and lines wrap at 80.
- [ ] Commits follow the [Ideal Git Commit Structure](https://github.com/lightningnetwork/lnd/blob/master/docs/code_contribution_guidelines.md#IdealGitCommitStructure).
- [ ] Any new logging statements use an appropriate subsystem and logging level.
- [ ]  [There is a change description in the release notes](https://github.com/lightningnetwork/lnd/tree/master/docs/release-notes), or `[skip ci]` in the commit message for small changes.

📝 Please see our  [Contribution Guidelines](https://github.com/lightningnetwork/lnd/blob/master/docs/code_contribution_guidelines.md) for further guidance.

### CODE REVIEW

File: funding/manager.go [hunk 1]
```diff
@@ -1469,15 +1482,24 @@ func (f *Manager) handleFundingOpen(peer lnpeer.Peer,
...
-	public := msg.ChannelFlags&lnwire.FFAnnounceChannel != 0
-	if public && scid {
+	case public && scid:
 		err = fmt.Errorf("option-scid-alias chantype for public " +
 			"channel")
 		log.Error(err)
 		f.failFundingFlow(peer, msg.PendingChannelID, err)
 		return
+
+	// The current variant of taproot channels can only be should only be
```
Comment (hieblmi) [inline funding/manager.go:1496]: nit: "can only be"

File: funding/manager.go [hunk 2]
```diff
@@ -2118,12 +2155,27 @@ func (f *Manager) continueFundingAccept(resCtx *reservationWithCtx,
...
 		FundingPoint:     *outPoint,
 	}
-	fundingCreated.CommitSig, err = lnwire.NewSigFromSignature(sig)
-	if err != nil {
-		log.Errorf("Unable to parse signature: %v", err)
-		f.failFundingFlow(resCtx.peer, pendingChanID, err)
-		return
+
+	// If this is a taproot channel, then we'll need to populate the musig2
+	// partial sig field instead of the regaulr commit sig field.
```
Comment (hieblmi) [inline funding/manager.go:2160]: nit: regular

File: funding/manager.go [hunk 3]
```diff
@@ -2226,21 +2288,34 @@ func (f *Manager) handleFundingCreated(peer lnpeer.Peer,
...
-		return
+	fundingSigned := &lnwire.FundingSigned{
+		ChanID: channelID,
 	}
 
-	fundingSigned := &lnwire.FundingSigned{
-		ChanID:    channelID,
-		CommitSig: ourCommitSig,
+	// For taproot channels, we'll need to send over a partial signature
+	// that includes the nonce along sidethe signature.
```
Comment (hieblmi) [inline funding/manager.go:2296]: nit: alongside the

File: funding/manager.go [hunk 4]
```diff
@@ -2882,6 +2978,38 @@ func (f *Manager) sendFundingLocked(completeChan *channeldb.OpenChannel,
 	}
 	fundingLockedMsg := lnwire.NewFundingLocked(chanID, nextRevocation)
 
+	// If this is a taproot channel, then we also need to send along our
```
Comment (hieblmi) [inline funding/manager.go:2981]: nit: 'also' or 'as well'.

File: rpcserver.go [hunk 5]
```diff
@@ -2119,11 +2121,39 @@ func (r *rpcServer) parseOpenChannelReq(in *lnrpc.OpenChannelRequest,
...
+		}
+
+		channelType = new(lnwire.ChannelType)
+		fv := lnwire.NewRawFeatureVector(
+			lnwire.SimpleTaprootChannelsRequired,
+		)
+
+		// TODO(roasbeef): no need for the rest as they're now
+		// implicit?
+
```
Comment (ellemouton) [inline rpcserver.go:2139]: are they? cant I want to support taproot chans but not want zero conf chans?

File: rpcserver.go [hunk 6]
```diff
@@ -2119,11 +2121,39 @@ func (r *rpcServer) parseOpenChannelReq(in *lnrpc.OpenChannelRequest,
...
+		}
+
+		*channelType = lnwire.ChannelType(*fv)
+
 	default:
 		return nil, fmt.Errorf("unhandled request channel type %v",
 			in.CommitmentType)
 	}
 
+	// TODO(roasbeef): make taproot the default chan type?
```
Comment (ellemouton) [inline rpcserver.go:2155]: perhaps we should wait a bit & give users ample warning that we will switch the default type?

File: lnrpc/lightning.proto [hunk 7]
```diff
@@ -1342,6 +1342,13 @@ enum CommitmentType {
     channel before its maturity date.
     */
     SCRIPT_ENFORCED_LEASE = 4;
+
+    /*
+    A channel that uses musig2 for the funding output, and the new tapscript
+    features where relevant.
+    */
+    // TODO(roasbeef): need script enforce mirror type for the above as well?
```
Comment (ellemouton) [inline lnrpc/lightning.proto:1350]: for my understanding, what is a "mirror type"?

**Review by ellemouton — COMMENTED** [review]
Short & Sweet 🍭  Looks good to go once the preceding ones are in and the build passes

**Review by Crypt-iQ — APPROVED** [review]
looks good

**Review by yyforyongyu — APPROVED** [review]
🤙

File: lnrpc/lightning.proto (hunk 7 above)
Comment (Roasbeef) [reply lnrpc/lightning.proto:1358]: So one for the script enforced lease scheme we use in Pool.

File: rpcserver.go (hunk 6 above)
Comment (Roasbeef) [reply rpcserver.go:2155]: Yep won't be the default type for a looong time

File: lncfg/protocol.go [hunk 8]
```diff
@@ -22,6 +22,10 @@ type ProtocolOptions struct {
 	// mini.
 	WumboChans bool `long:"wumbo-channels" description:"if set, then lnd will create and accept requests for channels larger chan 0.16 BTC"`
 
+	// TaprootChans should be set if we want to enable support for the
+	// experimental simple taproot chans commitment type.
+	TaprootChans bool `long:"taproot-chans" description:"if set, then lnd will create and accept requests for channels using the simple taproot commitment type"`
```
Comment (yyforyongyu) [inline lncfg/protocol.go:27]: also need this config in `protocol_integration.go`

**Review by yyforyongyu — CHANGES_REQUESTED** [review]

**Pull request merged**